src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs

```diff
@@ -467,8 +467,15 @@ private int CheckAndAdjustPositionForSpeculativeAttribute(int position)
 
         protected override IOperation GetOperationCore(SyntaxNode node, CancellationToken cancellationToken)
         {
+            if (!Root.FullSpan.Contains(node.FullSpan))
+            {
+                // given node must be sub node of this root.
+                return null;
```

Comment:
@cston @mavasani 
GetOperation is currently set to return null for invalid input. also, for something to throw ArgumentException, its input must be something user can validate before giving into the method. for this, I am not sure how user would know the given node is valid one or not.

I would rather change the method to TryGetOperation than throwing if naming is the issue.